Add dedicated page for running the notebooks

psobolewskiPhD opened the pull request:
Followup to: https://github.com/scipy-2024-image-analysis/tutorial/pull/3#issuecomment-2211148023
I think it may be worth explaining how to launch the notebooks.

psobolewskiPhD committed:
Add dedicated page for running the notebooks

psobolewskiPhD commented:
CC: @JoOkuma

JoOkuma commented:
@psobolewskiPhD, thanks, it's excellent. I have one comment.
The ```{important}... section isn't loading properly on GitHub markdown; it looks like generic code text.

![image](https://github.com/scipy-2024-image-analysis/tutorial/assets/21022743/d81da899-dc47-493f-a15e-2425e2d8b456)

psobolewskiPhD commented:
Oh yeah thanks for catching that!, 
It should look nice on the web, but for local you need:
https://github.com/executablebooks/jupyterlab-myst
Or I can just drop that formatting.
What do you think?

JoOkuma commented:
Maybe drop it? I think most participants will view it through GitHub or locally.
I'm using github to view it, I think they support fewer md features.

psobolewskiPhD commented:
I was figuring people would look at the webpage:
https://scipy-2024-image-analysis.github.io/tutorial/
Other pages already use admonitions... To make them look good local would just require the extra dependency. But yes, looking at the raw markdown in GitHub would show a code block.

JoOkuma commented:
@psobolewskiPhD That makes sense; I should have paid more attention to the webpage.
LGTM, I think you can merge it.

psobolewskiPhD committed:
add jupyterlab-myst to conda env

psobolewskiPhD committed:
add jupyterlab-myst to req.txt

psobolewskiPhD commented:
I added the dependency for local notebooks.
BTW Do y'all want a workflow to build the website as an artifact to download and preview locally?

JoOkuma commented:
I'm indifferent about having the website available locally. You can decide @psobolewskiPhD

lagru commented on tutorial/notebook_setup.md:
I think you need to include this file in the `_toc.yml` for it to show up.

lagru commented on tutorial/notebook_setup.md:
```diff
@@ -0,0 +1,71 @@
+# Launching the Jupyter notebooks
```
In case we include it, this should also be numbered like all the other files. Maybe

```suggestion
# Part 0.2: Launching the Jupyter notebooks
```

and update `install_dependencies.md` to "Part 0.1"?

lagru commented on tutorial/notebook_setup.md:
```diff
@@ -0,0 +1,71 @@
+If you cloned the workshop repository, then you already have the notebooks in the `tutorial` subfolder.
+If you have not then you can download the notebooks as follows:
+
+### Cloning via git
```
Is this intended to come before the `install_dependencies.md` file? Then we might have a problem, because our [tutorial page](https://cfp.scipy.org/2024/talk/PQMQ3K/) points at the `install_dependencies.md` as the first entry point. 

What you you think about inserting the content of this file as the first section in `install_dependencies.md`?

psobolewskiPhD replied:
For people to install the dependencies following the instructions they need the env.yaml so I copied these instructions to `install_dependencies.md`.
I dont think everyone will need the notebook setup instructions, so duplication seems fine?

lagru replied:
I'm somewhat against the duplication, because it makes for another paragraph newbies have to read through to decide if they need to follow it. 

Honestly, is there any reason not to include the rest of this content on launching the notebooks at the end of the `install_dependencies.md` too? Having this in another file doesn't really add much.

And launching the notebooks is something that we will demonstrate during the tutorial anyway.

psobolewskiPhD replied:
Sure!

lagru reviewed (commented):
Thanks for adding the additional explanation! With the concern about our tutorial instructions pointing at `install_dependencies.md` first, the README is somewhat invisible now. So we should definitely include the instructions on how to get the repo! 

Though, I think the ordering of instructions is somewhat confusing now. I'm not sure whether the new instructions are supposed to go before or after the existing `install_dependencies.md`. 

Anyway, I believe an order that makes sense would be 

- Get the repo with git or as a ZIP (essentially the README)
- Setup the environment and dependencies (`install_dependencies.md`)
- Starting / opening notebooks

With the concern about our tutorial instructions pointing at `install_dependencies.md` first, I'd recommend just including everything in that file? What do you think?

lagru commented on environment.yml:
```diff
@@ -5,6 +5,7 @@ dependencies:
   - python=3.12
   - jupyterlab
   - jupytext
+  - jupyterlab-myst
```
I'd be hesitant to tweak our dependencies this late, but as far as I'm aware it will not break anything. Users missing this will just not see nice admonitions..? So okay in my book. :)

psobolewskiPhD replied:
Yeah it's just a visual thing -- the existing notebooks already use admonitions, so may as well IMO?

lagru replied:
Go for it. :)

lagru commented:
> BTW Do y'all want a workflow to build the website as an artifact to download and preview locally?

I'd say go for it, if we had more time left. :sweat_smile:

psobolewskiPhD commented:
I think I forgot to add the new page to the ToC 😝

psobolewskiPhD committed:
add notebook setup to ToC and copy the clone/zip to install dependencies.

psobolewskiPhD committed:
clarify the install/clone/download

psobolewskiPhD commented:
OK, I added the new page to the toc and added the clone/download intructions in the `install_dependencies` page -- people at a minimum need the env.yaml.

lagru commented on tutorial/install_dependencies.md:
```diff
@@ -85,6 +85,37 @@ as well and you can skip to the next section.
+If you cloned the workshop repository, then you already have everything you
+ need to set up the tutorial environment, including the notebooks in the
+  `tutorial` subfolder. You can skip to the next section.  
+If you havn't cloned the repository, then at a minimum you will need the 
+  [conda environment file](https://github.com/scipy-2024-image-analysis/tutorial/raw/main/environment.yml) to set up the environment.
```
I don't really follow this sentence. I assume people that want the environment (and thus the env file) will also want the notebooks to run and follow the tutorial?

Maybe just scratch the sentence?

lagru commented on tutorial/install_dependencies.md:
````diff
@@ -85,6 +85,37 @@ as well and you can skip to the next section.
+### Cloning via git
+To clone the repository containing the tutorial materials to your computer, open
+your Terminal and navigate to the folder where you will download the course
+materials into. We recommend cloning the materials into your Documents folder,
+but you can choose another suitable location. 
+
+ ```bash
+ cd ~/Documents
+ ```
````
This is something that Windows users might trip on unexpectedly. I'd make the tradeoff, not include this and let users clone wherever they want?

psobolewskiPhD committed:
Update for review comments

psobolewskiPhD committed:
Move the new content to the existing install file

psobolewskiPhD committed:
Fix ToC

psobolewskiPhD commented:
@lagru I moved the new content to the existing file, but kept the 0.1 and 0.2 headings.
FYI I'm still en route to Tacoma (layover at the moment), so feel free to push to this branch, merge or close or whatever, as I will be unlikely to see any comments until sometime tomorrow.

lagru committed:
Cleanup / streamline headings

Two top-level headings weren't rendered correctly by jupyter-book in
the sidebar. Instead just rename the section to "Setup" which should be
more fitting.

Also make sure that all headings use the same mood: imperative.

lagru reviewed (approved):
Alright, I streamlined the headings somewhat in https://github.com/scipy-2024-image-analysis/tutorial/pull/8/commits/29602d12872ec3ef1a86ae439bf50d9c551e17f2 (see description).

Pretty happy with this now. Thanks for improving these instructions significantly! :D

I'll merge now but please feel free to comment I've you see room for improvement or disagree with something. I'll be happy to start a follow-up PR.
